service/src/main/kotlin/io/provenance/digitalcurrency/consortium/stream/EventStreamConsumer.kt

```diff
@@ -63,6 +79,110 @@ class EventStreamConsumer(
+
+    private fun handleCoinMovementEvents(blockHeight: Long, batch: EventBatch) {
+        // TODO (steve) is there a grpc endpoint for this?
+        val block = rpcClient.fetchBlock(blockHeight).block
+
+        // SC Mint events denote the "on ramp" for a bank user to get coin
+        batch.mints(provenanceProperties.contractAddress)
+            .filter { it.withdrawAddress.isNotEmpty() && it.memberId.isNotEmpty() }
+            .forEach { event ->
+                log.debug("Mint - tx: $${event.txHash} member: ${event.memberId} withdrawAddr: ${event.withdrawAddress} amount: ${event.amount} denom: ${event.withdrawDenom}")
+
+                // TODO (steve) implement caching
+                // TODO (steve) move to getAttributes by tag name
+                val toAddressBankUuid = pbcService.getAttributes(event.withdrawAddress).bankUuid()
+
+                // persist a record of this transaction if the to address has this bank's attribute, the from address will be the SC address
+                if (toAddressBankUuid != null) {
+                    // TODO (steve) change to upsert
+                    transaction {
+                        CoinMovementRecord.insert(
+                            // TODO (steve) txhash is not unique in the general case so we have to fix this - but it works for now considering
+                            // we don't batch calls from this service at all
+                            txHash = event.txHash,
+                            fromAddress = event.memberId,
+                            fromAddressBankUuid = null,
+                            toAddress = event.withdrawAddress,
+                            toAddressBankUuid = toAddressBankUuid,
+                            blockHeight = event.height,
+                            blockTime = OffsetDateTime.parse(block.header.time),
+                            amount = event.amount,
+                            denom = event.withdrawDenom,
+                            type = MINT,
+                        )
+                    }
+                }
+            }
+
+        // SC Transfer events denote the "off ramp" for a bank user to redeem coin when the recipient is the bank address
+        batch.transfers(provenanceProperties.contractAddress)
+            .filter { it.sender.isNotEmpty() && it.recipient.isNotEmpty() }
+            .forEach { event ->
+                log.debug("SC Transfer - tx: ${event.txHash} sender: ${event.sender} recipient: ${event.recipient} amount: ${event.amount} denom: ${event.denom}")
+
+                // TODO (steve) implement caching
+                // TODO (steve) move to getAttributes by tag name
+                val fromAddressBankUuid = pbcService.getAttributes(event.sender).bankUuid()
+
+                // persist a record of this transaction if either the from or the to address has this bank's attribute
+                if (event.recipient == pbcService.managerAddress && fromAddressBankUuid != null) {
```

Comment:
i was under assumption we had to store all transfers even if we're not just sending to the bank?

Reply:
oh nvm thats the markerTransfers below :doh:

Reply:
Yeah, the function names don't show this fact well, but `batch.transfers(..)` is SC transfer events, which when filtered like this one is, amounts to BURNs (transfers from a user to a dcc bank). The `batch.markerTransfer()` is `EventMarkerTransfer` events and due to the filtering there it is strictly user to user transfers where none, one, or both sides is a nycb user.